fv/ipip_test.go

```diff
@@ -250,6 +251,106 @@ var _ = infrastructure.DatastoreDescribe("IPIP topology before adding host IPs t
 			cc.ExpectSome(w[1], w[0])
 			cc.CheckConnectivity()
 		})
+
+		Context("with policy allowing port 8055", func() {
+			BeforeEach(func() {
+				tcp := numorstring.ProtocolFromString("tcp")
+				udp := numorstring.ProtocolFromString("udp")
+				p8055 := numorstring.SinglePort(8055)
+				policy := api.NewGlobalNetworkPolicy()
+				policy.Name = "allow-8055"
+				policy.Spec.Ingress = []api.Rule{
+					{
+						Protocol: &udp,
+						Destination: api.EntityRule{
+							Ports: []numorstring.Port{p8055},
+						},
+						Action: api.Allow,
+					},
+					{
+						Protocol: &tcp,
+						Destination: api.EntityRule{
+							Ports: []numorstring.Port{p8055},
+						},
+						Action: api.Allow,
+					},
+				}
+				policy.Spec.Egress = []api.Rule{
+					{
+						Protocol: &udp,
+						Destination: api.EntityRule{
+							Ports: []numorstring.Port{p8055},
+						},
+						Action: api.Allow,
+					},
+					{
+						Protocol: &tcp,
+						Destination: api.EntityRule{
+							Ports: []numorstring.Port{p8055},
+						},
+						Action: api.Allow,
+					},
+				}
+				policy.Spec.Selector = fmt.Sprintf("has(host-endpoint)")
+				_, err := client.GlobalNetworkPolicies().Create(utils.Ctx, policy, utils.NoOptions)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			// Test each of these cases separately, to avoid the possibility of one case
+			// setting up conntrack state that allows a different case to pass.
+			It("allows host0 to remote host-networked workload", func() {
```

Comment:
Might have missed something but I'm not sure this tests the case that was broken before.  I think you need a test that blocks traffic on eth0 but asserts that traffic to a remote pod still gets through (because the eth0 hep doesn't apply to the IPIP/vxlan interface).

Reply:
The test for the previously broken case is in hostendpoints_test.go, line 118 (`expectHostToOwnPodViaServiceTraffic`).

This one is arguably superfluous, and perhaps I should remove it.  It's host to host and - IIUC - doesn't use the IP-IP overlay.  My purpose here (and similarly in vxlan_test.go) was to verify the need (or not) for the IPIP/VXLAN failsafe rules in StaticManglePostrouting, and for that I only really need the "allows host0 to remote Calico-networked workload via service IP" cases.

WDYT?  Should I remove the other cases that I've added?  (They may well duplicate testing that we already have elsewhere.)

Reply:
As agreed, I've removed the superfluous cases now.